### PR TITLE
Fix tint image bug which leads to image upside down

### PR DESCRIPTION
**Pull request by zhongwuzw**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [ ] I have added the required tests to prove the fix/feature I am adding
* [ ] I have updated the documentation (if necessary)
* [ ] I have run the tests and they pass
* [ ] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: ...

### Pull Request Description

Same issue as #2673 .

### CODE REVIEW

File: SDWebImage/UIImage+Transform.m [hunk 1]
```diff
@@ -349,7 +349,7 @@ - (UIImage *)sd_tintedImageWithColor:(UIColor *)tintColor {
     
     SDGraphicsBeginImageContextWithOptions(size, NO, scale);
     CGContextRef context = SDGraphicsGetCurrentContext();
-    CGContextDrawImage(context, rect, self.CGImage);
+    [self drawInRect:rect];
     CGContextSetBlendMode(context, blendMode);
```
Comment (bpoplauschi) [inline SDWebImage/UIImage+Transform.m]: this method used to work with CoreGraphics functions only. I would be curious to know what other change in the lib has caused this upside down issue.
I'm not saying it's a bad fix, but it sure looks like it doesn't match the rest of the method body

Comment (dreampiggy) [reply SDWebImage/UIImage+Transform.m]: Actually, this code can be placed like this:

```objective-c
#if SD_UIKIT || SD_WATCH
CGContextScaleCTM(context, 1.0, -1.0);
CGContextTranslateCTM(context, 0, -size.height);
#endif 
CGContextDrawImage(context, rect, self.CGImage);
```

But using that `drawInRect:` seems a little convienet than Core Graphics APIs. So I think it's OK.

Comment (dreampiggy) [reply SDWebImage/UIImage+Transform.m]: @bpoplauschi We update all test cases to ensure this does not cause bug in the future. Currently only this and `sd_roundedCornerImageWithRadius:` contains the same issue.

Comment (bpoplauschi) [reply SDWebImage/UIImage+Transform.m]: got it. thanks @dreampiggy

**Pull request merged**